test/hooks.test.js

```diff
@@ -2591,16 +2591,24 @@ test('preSerialization hooks should support encapsulation', t => {
 })
 
 test('onRegister hook should be called / 1', t => {
-  t.plan(3)
+  t.plan(4)
   const fastify = Fastify()
 
+  const pluginOpts = { prefix: 'hello', custom: 'world' }
   fastify.register((instance, opts, next) => {
     next()
-  })
+  }, pluginOpts)
 
-  fastify.addHook('onRegister', instance => {
+  let first = true
+  fastify.addHook('onRegister', (instance, opts) => {
     // duck typing for the win!
     t.ok(instance.addHook)
+    if (first) {
+      // the first call of the onRegister is the main fastify instance, not the registered ones
```

Comment:
What you think about send a `opts.first` or something else from register main fastify instance? Can be more clear :smile:

Reply:
I should change the opts object send by the user and it could be harmful